Epsilon.Host.Frontend/src/components/competence/CompetenceDocument.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+	<div v-if="submissions" class="competence-document">
+		<KpiTable :submissions="submissions" :outcomes="allOutcomes" />
+		<div></div>
+		<div></div>
+		<CompetenceProfile
+			v-if="hboIDomain"
+			:submissions="submissions"
+			:domain="hboIDomain" />
+		<div></div>
+		<div></div>
+		<KpiMatrix
+			v-if="hboIDomain"
+			:submissions="submissions"
+			:domain="hboIDomain" />
+	</div>
+</template>
+
+<script lang="ts" setup>
+import KpiMatrix from "~/components/competence/KpiMatrix.vue"
+import KpiTable from "~/components/competence/KpiTable.vue"
+import CompetenceProfile from "~/components/competence/CompetenceProfile.vue"
+import type {
+	LearningDomain,
+	LearningDomainSubmission,
+	LearningDomainOutcome,
+} from "~/api.generated"
+
+const props = defineProps<{
+	submissions: LearningDomainSubmission[]
+}>()
+
+const api = useApi()
+
+const hboIDomain = ref<LearningDomain | null>(null)
+const personalDevelopmentDomain = ref<LearningDomain | null>(null)
+
+onMounted(async () => {
+	hboIDomain.value = (await api.learning.domainDetail("hbo-i-2018")).data
+	personalDevelopmentDomain.value = (
+		await api.learning.domainDetail("pd-2020-bsc")
+	).data
+})
+
+const allOutcomes = computed<LearningDomainOutcome[]>(() =>
+	props.submissions.flatMap((submission) =>
+		submission.results!.map((result) => result.outcome!)
+	)
+)
+</script>
+
+<style scoped>
+.competence-document {
```

Comment:
The question remains, do you what all of this does? 
I don't. The looks very complex css for what?

Reply:
It was taken from the PerformanceDashboard page. Have taken a second look at it and took a bit of CSS out.